client/components/Username.vue

```diff
@@ -72,7 +77,18 @@ export default defineComponent({
 
 		const store = useStore();
 
+		const userModes = {
+			"~": "owner",
+			"&": "admin",
+			"!": "admin",
+			"@": "op",
+			"%": "half-op",
+			"+": "voice",
+			"": "normal",
+		};
```

Comment:
All of those are configurable on the ircd and hence specific to a net, you can't just assume that they are layered this way iirc.
The only thing guaranteed is op and voice.

Further, this duplicates similar functionality in https://github.com/thelounge/thelounge/blob/a8be84028c2aeea39fda5adea446aaaaca66aa2c/client/js/helpers/contextMenu.ts#L362

Reply:
I actually copied this over from an existing file:

https://github.com/thelounge/thelounge/blob/a8be84028c2aeea39fda5adea446aaaaca66aa2c/client/components/ChatUserList.vue#L66

Should I just use the one you referenced instead? Also, do you want me to put it inside constants as well to prevent redundancy?

https://github.com/thelounge/thelounge/blob/a8be84028c2aeea39fda5adea446aaaaca66aa2c/client/js/constants.ts

Reply:
/me screams

Well, if you can clean it up please do.
They should probably all share *one* implementation